src/common/dobjcmn.cpp

```diff
@@ -530,21 +553,58 @@ bool wxHTMLDataObject::SetData(size_t WXUNUSED(len), const void *buf)
     // Windows and Mac always use UTF-8, and docs suggest GTK does as well.
     wxString html = wxString::FromUTF8(static_cast<const char*>(buf));
 
-#ifdef __WXMSW__
-    // To be consistent with other platforms, we only add the Fragment part
-    // of the Windows HTML clipboard format to the data object.
-    int fragmentStart = html.rfind("StartFragment");
-    int fragmentEnd = html.rfind("EndFragment");
+    // Although some applications such as Word may pass the correctly-sized buffer, let's
+    // ensure we can read the HTML even if there's garbage at the end. For example,
+    // on Windows, LibreOffice includes extraneous terminating characters, as can be seen via a
+    // clipboard viewer such as Free Clipboard Viewer.
```

Comment:
Hi Vadim, I'm using LO 7.0.3.1 on Windows. Type some text e.g. "Some text", copy to the clipboard, then look at Free Clipboard Viewer 3.0, the "HTML Format" type. There's ".." at the end after the close HTML tag. Whether it's guaranteed to be correct UTF-8 I don't know, but I think there's a fair chance this could be garbage, and if LO does it, possibly other applications might do it.

Reply:
Couldn't this ".." be just `"\r\n"`? Can you see the hex values of the bytes corresponding to it?